src/common/components/build-row/index.js

```diff
@@ -0,0 +1,55 @@
+import React, { PropTypes } from 'react';
+import { Link } from 'react-router';
+import moment from 'moment';
+
+import styles from './buildRow.css';
+
+const BuildRow = (props) => {
+
+  const {
+    account,
+    repo,
+    architecture,
+    buildId,
+    duration,
+    status,
+    statusMessage,
+    dateStarted,
+    dateCompleted
+  } = props;
+
+  return (
+    <div className={ `${styles.buildRow} ${styles[status]}` }>
+      <div className={ styles.item }><Link to={`/${account}/${repo}/builds/${buildId}`}>{`#${buildId}`}</Link> {statusMessage}</div>
+      <div className={ styles.item }>
+        {architecture}
+      </div>
+      <div className={ styles.item }>
+        {moment.duration(duration).humanize()}
+      </div>
+      <div className={ styles.item }>
+        <span title={moment(dateCompleted || dateStarted).format('DD-MM-YYYY HH:mm:ss')}>{ moment(dateCompleted || dateStarted).fromNow() }</span>
```

Comment:
I removed `dateCompleted`. I guess showing one common started date will be less confusing.
Later if we want we may show both dates in build details, but I guess date started and duration give enough information.